pgmq/extension/src/partition.rs

```diff
@@ -1,54 +1,117 @@
+use pgx;
+use pgx::prelude::*;
+
 use pgmq_crate::{
     errors::PgmqError,
     query::{check_input, create_archive, create_index, create_meta, insert_meta, TABLE_PREFIX},
 };
 
-// for now, put pg_partman in the public schema
+// for now, put pg_partman in the public PGMQ_SCHEMA
 const PARTMAN_SCHEMA: &str = "public";
-const SCHEMA: &str = "public";
+const PGMQ_SCHEMA: &str = "public";
 
-pub fn init_partitioned_queue(name: &str, partition_size: i64) -> Result<Vec<String>, PgmqError> {
+pub fn init_partitioned_queue(
+    name: &str,
+    partition_interval: &str,
+    retention_interval: &str,
+) -> Result<Vec<String>, PgmqError> {
     check_input(name)?;
+    let partition_col = map_partition_col(partition_interval);
     Ok(vec![
         create_meta(),
-        create_partitioned_queue(name)?,
-        create_partitioned_index(name)?,
+        create_partitioned_queue(name, &partition_col)?,
+        create_partitioned_index(name, &partition_col)?,
         create_index(name)?,
         create_archive(name)?,
-        create_partman(name, partition_size)?,
+        create_partitioned_table(name, &partition_col, partition_interval)?,
         insert_meta(name)?,
+        set_retention_config(name, retention_interval)?,
     ])
 }
 
-pub fn create_partitioned_queue(queue: &str) -> Result<String, PgmqError> {
+/// maps the partition column based on partition_interval
+fn map_partition_col(partition_interval: &str) -> String {
+    match partition_interval.parse::<i32>() {
+        Ok(_) => "msg_id".to_owned(),
+        Err(_) => "enqueued_at".to_owned(),
+    }
+}
+
```

Comment:
when `pgmq_create()` is a integer, then we need to partition by `msg_id`. Otherwise, assuming its a date and partitioning by `enqueued_at`